src/dpp/cluster.cpp

```diff
@@ -70,6 +60,38 @@ template<typename T> std::function<void(const T&)> make_intent_warning(cluster*
 	};
 }
 
+template <build_type BuildType>
+bool validate_configuration() {
+#ifdef _DEBUG
+	constexpr build_type expected = build_type::debug;
+#else
+	constexpr build_type expected = build_type::release;
+#endif
+#ifdef _WIN32
+	if constexpr (BuildType != build_type::universal && BuildType != expected) {
+		MessageBox(
+			nullptr,
+			"Mismatched Debug/Release configurations between project and dpp.dll.\n"
+			"Please ensure both your program and the D++ DLL file are both using the same configuration.\n"
+			"The program will now terminate.",
+			"D++ Debug/Release mismatch",
+			MB_OK | MB_ICONERROR
+		);
+		/* Use std::runtime_rror here because dpp exceptions use std::string and that would crash when catching, because of ABI */
+		throw std::runtime_error("Mismatched Debug/Release configurations between project and dpp.dll");
+	}
+	return true;
```

Comment:
does it need to return bool at all? either it's valid and returns or it's not and it throws. there is no false 😁

Reply:
It needs to return *something*. It's in conjunction with the global variables at https://github.com/brainboxdotcc/DPP/pull/980/files#diff-904a828bab8b0e5e1fa5ab9a9f230d8675fbedc077dac48d80bacb56b2e3cc70R106, they need to be initialized with *something* so that the function is called on program startup.

Reply:
what stops is_valid_config being optimised away as unused? I guess it's never been an issue, I'm just curious

Reply:
When linking D++ dynamically, the compiler cannot determine that a call to `validate_configuration` has no side effects, therefore it will not optimize it away, as that would potentially violate the as-if rule (a compiler must maintain observable behavior when optimizing)

Specifically: https://eel.is/c++draft/basic.stc.static#2